src/CoreWCF.Primitives/src/CoreWCF/Configuration/ServiceBuilder.cs

```diff
@@ -15,9 +16,16 @@ internal class ServiceBuilder : CommunicationObject, IServiceBuilder
         private readonly IDictionary<Type, IServiceConfiguration> _services = new Dictionary<Type, IServiceConfiguration>();
         private readonly TaskCompletionSource<object> _openingCompletedTcs = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously);
 
-        public ServiceBuilder(IServiceProvider serviceProvider)
+        public ServiceBuilder(IServiceProvider serviceProvider, IApplicationLifetime appLifetime)
         {
             ServiceProvider = serviceProvider;
+            appLifetime.ApplicationStarted.Register(() =>
+            {
+                if (State == CommunicationState.Created)
+                {
+                    OpenAsync().GetAwaiter().GetResult();
```

Comment:
Do you need to call OpenAsync here or can it be started and awaited elsewhere?

Reply:
Thinking about it, it doesn't actually need to be waited here at all. This is for the case when IIS hosted and there isn't a wrapping IServer to ensure OpenAsync is called. So this is a fallback so that if nothing previously called OpenAsync, we make sure it gets called but we won't see the exception. I'll take out the waiting and use a discard.